### PR TITLE
Use `ensure_can_withdraw` instead of balance check

### PR DESCRIPTION
**Pull request by zjb0807**
Fix issues #39.
Use `ensure_can_withdraw` instead of balance check in modules/synthetic-protocol/src/lib.rs

### CODE REVIEW

**Comment by zjb0807** [comment]
https://github.com/laminar-protocol/laminar-chain/blob/34e4f8e2b0f3ff8fa610521406cf481dc4a29aca/modules/synthetic-protocol/src/lib.rs#L345-L348
This can't use ensure_can_withdraw instead of balance check, because the condition is `>` not `>=`.

File: modules/synthetic-protocol/src/lib.rs [hunk 1]
```diff
@@ -242,11 +236,10 @@ impl<T: Trait> Module<T> {
...
 
-		ensure!(
-			T::CollateralCurrency::balance(&<SyntheticTokens<T>>::account_id())
-				>= redeemed_collateral + pool_refund_collateral,
-			Error::<T>::NotEnoughLockedCollateralAvailable,
-		);
+		T::CollateralCurrency::ensure_can_withdraw(
+			&<SyntheticTokens<T>>::account_id(),
+			redeemed_collateral + pool_refund_collateral,
+		)?;
```
Comment (shaopengw) [inline modules/synthetic-protocol/src/lib.rs]: Map the error to `Error::<T>::NotEnoughLockedCollateralAvailable`. The fact that collateral is locked in synthetic tokens module account should not be exposed to user.

File: modules/synthetic-protocol/src/tests.rs [hunk 2]
```diff
@@ -326,7 +326,7 @@ fn redeem_fails_if_not_enough_locked_collateral() {
 
 			assert_noop!(
 				redeem_ausd(ALICE, SyntheticCurrency::balance(&ALICE)),
-				Error::<Runtime>::NotEnoughLockedCollateralAvailable,
+				Error::<Runtime>::BalanceTooLow,
```
Comment (shaopengw) [inline modules/synthetic-protocol/src/tests.rs]: Same here, map to `NotEnoughLockedCollateralAvailable`.

**Comment by xlc** [comment]
> https://github.com/laminar-protocol/laminar-chain/blob/34e4f8e2b0f3ff8fa610521406cf481dc4a29aca/modules/synthetic-protocol/src/lib.rs#L345-L348
> 
> 
> This can't use ensure_can_withdraw instead of balance check, because the condition is `>` not `>=`.

@shaopengw any reason this is `>` not `>=`?

**Comment by shaopengw** [comment]
I think it's a typo, should be `>=`.

File: modules/synthetic-protocol/Cargo.toml [hunk 3]
```diff
@@ -21,11 +21,11 @@ orml-utilities = { path = "../../orml/utilities", default-features = false }
 module-primitives = { path = "../primitives", default-features = false }
 traits = { package = "module-traits", path = "../traits", default-features = false }
 synthetic-tokens = { package = "module-synthetic-tokens", path = "../synthetic-tokens", default-features = false }
+orml-tokens = { path = "../../orml/tokens", default-features = false }
```
Comment (xlc) [inline modules/synthetic-protocol/Cargo.toml]: tokens and currencies are dev-dependencies not dependencies

should be just this in dev-dependencies, also no need the std entries in features

```
orml-tokens = { path = "../../orml/tokens" }
orml-currencies = { path = "../../orml/currencies" }
```

**Pull request merged**